dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -152,6 +152,25 @@ public void onFailure(Throwable throwable) {
         }
     }
 
+    private enum  RateLimitedResponse implements Response {
+        INSTANCE;
+
+        @Override
+        public InputStream body() {
+            return InputStream.nullInputStream();
+        }
+
+        @Override
+        public int code() {
+            return 429;
```

Comment:
this is different from unavailable

Reply:
yeah I think this is representative of what you want. you're seeing this because you've reached your concurrency limit and you'd expect that downstream service to response with a 429 if you were to execute a request. You also wouldn't expect going to another server to help much here because it's the underlying service that's overloaded.